module/zfs/vdev.c

```diff
@@ -1515,6 +1515,24 @@ vdev_metaslab_group_create(vdev_t *vd)
 	}
 }
 
+void
+vdev_update_nonallocating_space(vdev_t *vd, boolean_t add)
+{
+	spa_t *spa = vd->vdev_spa;
+	if (vd->vdev_mg->mg_class == spa_normal_class(spa)) {
+		uint64_t raw_space = vd->vdev_stat.vs_space -
+		    metaslab_group_get_space(vd->vdev_log_mg);
```

Comment:
Since device removal on completion frees vdev spacemaps, I wonder whether `mg_metaslab_tree` used by `metaslab_group_get_space()` will be populated after removal.

Or on opposite, if the trees are valid, can't we then use `metaslab_group_get_space(vd->vdev_mg)` instead?

Reply:
The call to `vdev_metaslab_fini` is what clears out the metaslab groups (`metaslab_group_remove` is called from `metaslab_fini` is called from `vdev_metaslab_fini`), and that is called just after the call to `vdev_update_nonallocating_space` in `vdev_remove_complete`. `vdev_metaslab_fini` is also what updates the `vs_space` metric (via `metaslab_space_update` from `metaslab_fini`), so this already had to be before that, which is why nothing had to be moved.